### ISSUE TITLE
Crash when requesting permissions

### ISSUE THREAD

**Issue by samtstern** [googlesamples/easypermissions#65]
Using version `0.2.1` I am sometimes seeing this crash in production:

```
Exception java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=101, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS VirtualScreenParam=Params{mDisplayId=-1, null, mFlags=0x00000000)} (has extras) }} to activity {com.google.android.apps.santatracker/com.google.android.apps.santatracker.presentquest.MapsActivity}: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
android.app.ActivityThread.deliverResults (ActivityThread.java:4925)
android.app.ActivityThread.handleSendResult (ActivityThread.java:4968)
android.app.ActivityThread.access$1600 (ActivityThread.java:222)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1849)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:158)
android.app.ActivityThread.main (ActivityThread.java:7229)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1230)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1120)
```

```
Caused by java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
android.support.v4.app.FragmentManagerImpl.checkStateLoss (FragmentManager.java:1533)
android.support.v4.app.FragmentManagerImpl.enqueueAction (FragmentManager.java:1551)
android.support.v4.app.BackStackRecord.commitInternal (BackStackRecord.java:696)
android.support.v4.app.BackStackRecord.commit (BackStackRecord.java:662)
android.support.v4.app.DialogFragment.show (DialogFragment.java:139)
pub.devrel.easypermissions.EasyPermissions.showRationaleDialogFragmentCompat (EasyPermissions.java:176)
pub.devrel.easypermissions.EasyPermissions.requestPermissions (EasyPermissions.java:149)
pub.devrel.easypermissions.EasyPermissions.requestPermissions (EasyPermissions.java:104)
```

**Comment by samtstern** [comment]
It's not clear to me if this is a bug in EasyPermissions or in the application (very likely the latter) but it will be good to investigate and provide guidance to others.

**Comment by youfacepalm** [comment]
Would it be because you are trying to open a DialogFragment within onActivityResult() method?

**Comment by samtstern** [comment]
@youfacepalm is doing so a known issue?  It's also worth nothing that we only saw this crash on API 23 devices, not 24 or 25 devices.

**Comment by youfacepalm** [comment]
Yes. Please check [this](http://stackoverflow.com/a/18345899/3533289) StackOverflow answer by Alex Lockwood as he explains why it is a bad idea to commit fragment transaction in onActivityResult().

There is also another AOSP [bug](https://code.google.com/p/android/issues/detail?id=207269) being tracked related to DialogFragments throwing IllegalStateException. It primarily happens when DialogFragment is committed within an OnClickListener so I am not sure if it is relevant in this case

**Comment by samtstern** [comment]
This should be fixed by #75 since we no longer use a `DialogFragment` directly but rather launch a transparent 'host' Activity to contain the `DialogFragment`.
